pkg/runtime/reconciler.go

```diff
@@ -268,49 +268,113 @@ func (r *resourceReconciler) Sync(
 					// there is some changes available in the latest.RuntimeObject()
 					// (example: ko.Status.Conditions) which have been
 					// updated in the resource
-					// Thus, patchResource() call should be made here
-					_ = r.patchResource(ctx, desired, latest)
+					// Thus, patchResourceStatus() call should be made here
+					_ = r.patchResourceStatus(ctx, desired, latest)
 				}
 				return err
 			}
 			rlog.Info("updated resource")
 		}
 	}
-	err = r.patchResource(ctx, desired, latest)
-	if err != nil {
-		return err
-	}
-	for _, condition := range latest.Conditions() {
-		if condition.Type == ackv1alpha1.ConditionTypeResourceSynced {
-			if condition.Status == corev1.ConditionTrue {
-				if duration := r.rmf.RequeueOnSuccessSeconds(); duration > 0 {
+	if ackcompare.IsNotNil(latest) {
+		err = r.patchResource(ctx, desired, latest)
+		if err != nil {
+			return err
+		}
+		for _, condition := range latest.Conditions() {
+			if condition.Type == ackv1alpha1.ConditionTypeResourceSynced {
+				if condition.Status == corev1.ConditionTrue {
+					if duration := r.rmf.RequeueOnSuccessSeconds(); duration > 0 {
+						rlog.Debug(
+							"requeueing resource after resource synced condition true",
+						)
+						return requeue.NeededAfter(nil, time.Duration(duration)*time.Second)
+					}
+				} else {
 					rlog.Debug(
-						"requeueing resource after resource synced condition true",
+						"requeueing resource after finding resource synced condition false",
 					)
-					return requeue.NeededAfter(nil, time.Duration(duration)*time.Second)
+					return requeue.NeededAfter(
+						ackerr.TemporaryOutOfSync, requeue.DefaultRequeueAfterDuration)
```

Comment:
nit: Inverting L285 check and using a `continue` statement will reduce the number of nested code scopes/bocks